### ISSUE TITLE
Training aways converges after 2 or 3 epochs

### ISSUE THREAD

**Issue by GabrielHoffman** [davek44/Basset#37]
Do you have any suggestions on exploring the hyper-parameter space in terms of learning rate, etc.  My concern is that using basset the prediction error on the validation set always converges after 2 or 3 epochs.  My guess is that selecting better hyper-parameter values will improve the prediction error.  Have you encountered this issue?

Thanks,
- Gabriel

**Comment by davek44** [comment]
Hi Gabriel,

Sorry I missed your comment here. My Harvard email turned off, and I just realized I wasn't getting notifications. 

If you're willing to put some time in to set it up, Spearmint (https://github.com/HIPS/Spearmint) is a great option for performing hyper-parameter searches. It sounds like your models are training quickly, so maybe you can perform a grid search in a reasonable amount of time, too.

Best,
Dave

**Comment by davek44** [comment]
@suragnair Yes, I'm pretty sure that's true. I can check if you need a definitive answer. In general, it's pretty robust to small changes to the model architecture, and more sensitive to optimization parameters.

**Comment by GabrielHoffman** [comment]
Hi @suragnair, I used the default model architecture in my current application, but I hope to get all to a more thorough search soon
